validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/KeystoresValidatorKeyProvider.java

```diff
@@ -46,12 +52,35 @@
 
     StatusLogger.STATUS_LOG.loadingValidators(keystorePasswordFilePairs.size());
     // return distinct loaded key pairs
-    return keystorePasswordFilePairs.stream()
-        .parallel()
-        .map(pair -> loadBLSPrivateKey(pair.getLeft(), loadPassword(pair.getRight())))
-        .distinct()
-        .map(privKey -> new BLSKeyPair(BLSSecretKey.fromBytes(privKey)))
-        .collect(toList());
+
+    final ExecutorService executorService =
+        Executors.newFixedThreadPool(Math.min(4, Runtime.getRuntime().availableProcessors()));
```

Comment:
This is a bit of a "magic number", though probably safe. Each Scrypt keystore takes about 300MB, so four is parallel is 1200mb, which is as low as we realistically might want to set the heap size, I guess.

For reference, I played with some code like this, which works well, but didn't manage to convince myself that it was a good idea:

```
    // This is a guess. Actual parameters seem to be in tech.pegasys.signers.bls.keystore.model.ScryptParam.java
    // final long memoryPerThread = 300_000_000L;
    final long memoryPerThread = 128L * 262_144L * 8L;
    final long allocatedMemory = (Runtime.getRuntime().totalMemory()-Runtime.getRuntime().freeMemory());
    final long presumableFreeMemory = Runtime.getRuntime().maxMemory() - allocatedMemory;
    final int nThreads = (int) Math.max(1, Math.min(Runtime.getRuntime().availableProcessors(), (presumableFreeMemory / memoryPerThread)));

    ForkJoinPool forkJoinPool = null;

    try {

      System.out.println("Available processors: " + Runtime.getRuntime().availableProcessors());
      System.out.println("Using " + nThreads + " threads to load validator keys");
      forkJoinPool = new ForkJoinPool(nThreads);

      StatusLogger.STATUS_LOG.loadingValidators(keystorePasswordFilePairs.size());
      // return distinct loaded key pairs
      return forkJoinPool.submit(() -> keystorePasswordFilePairs.stream()
              .parallel()
              .map(pair -> loadBLSPrivateKey(pair.getLeft(), loadPassword(pair.getRight())))
              .distinct()
              .map(privKey -> new BLSKeyPair(BLSSecretKey.fromBytes(privKey)))
              .collect(toList())).get();
etc...
```